sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -4363,16 +4363,17 @@ atomic_update(T *p, simd<Toffset, N> byte_offset, simd<T, N> src0,
 /// @return A vector of the old values at the memory locations before the
 ///   update.
 ///
-template <atomic_op Op, typename T, int N, typename Toffset,
-          typename RegionTy = region1d_t<T, N, 1>,
+template <atomic_op Op, typename T, int N, typename Toffset, typename ViewedTy,
+          typename RegionTy = region1d_t<ViewedTy, N, 1>,
           typename PropertyListT =
               ext::oneapi::experimental::detail::empty_properties_t>
 __ESIMD_API std::enable_if_t<
     __ESIMD_DNS::get_num_args<Op>() == 1 &&
         ext::oneapi::experimental::is_property_list_v<PropertyListT>,
     simd<T, N>>
-atomic_update(T *p, simd<Toffset, N> byte_offset, simd_view<T, RegionTy> src0,
-              simd_mask<N> mask, PropertyListT props = {}) {
+atomic_update(T *p, simd<Toffset, N> byte_offset,
+              simd_view<ViewedTy, RegionTy> src0, simd_mask<N> mask,
```

Comment:
We agreed not to have `simd_view` overloads for src0 and src1. They seemed redundant because, of implicit conversion from simd_view to simd.
but we need to keep `simd_view` overloads for `byte_offset` as the implicit conversion doesn't work for 'byte_offset` when it is `simd_view` - cannot match simd_view to simd error.